### DISCUSSION TITLE
Device can't connect to dockerhub

### DISCUSSION THREAD

**Discussion by AmarRobo** [AxisCommunications/acap-computer-vision-sdk-examples#124] [General] [answered]
I have an error while running the object-detector-python example. I am running on an ARTPEC-8 camera with TLS= NO option. 

ERROR: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on [::1]:53: read udp [::1]:40249->[::1]:53: read: connection refused

![ErrorwithOBjecDetPython_docker](https://user-images.githubusercontent.com/27479245/212019096-839bad9d-244e-4af2-9544-6a2854c4856a.png)

I am not able to understand the error. Can you help to find possible reasons for the error ?

**Comment by Corallo** [comment, marked as answer]
Hello, @amararjun 

The camera can't reach the docker registry 
Another user had your same issue, you can find how to fix it [here](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/35#discussioncomment-2897104).

> **Note**
> That user was using an older version of the acap-runtime, you want to pull instead the image that the camera is failing to pull in your error message. axisecp/acap-runtime:1.2.0-aarch64-containerized

**Reply by AmarRobo** [reply]
@Corallo    Thank you for the reply!
I did ` docker save axisecp/acap-runtime:1.2.0-aarch64-containerized | docker  -H tcp://$DEVICE_IP:$DOCKER_PORT load ` and it worked.
I now have another error while running the example program ; with accessing the shared library of acap runtime

`
docker-compose --host tcp://$DEVICE_IP:$DOCKER_PORT --file static-image.yml --env-file ./config/env.$ARCH.$CHIP up
`

gives following error :

WARNING: The INFERENCE_SERVER_COMMAND variable is not set. Defaulting to a blank string.
Starting object-detector-python_object-detector-python_1 ... done
Starting object-detector-python_inference-server_1       ... done
Starting object-detector-python_acap_dl-models_1         ... done
Attaching to object-detector-python_inference-server_1, object-detector-python_object-detector-python_1, object-detector-python_acap_dl-models_1
acap_dl-models_1          | COPYRIGHT
acap_dl-models_1          | coco_labels.txt
acap_dl-models_1          | ssd_mobilenet_v2_coco_quant_postprocess.tflite
acap_dl-models_1          | ssd_mobilenet_v2_coco_quant_postprocess_edgetpu.tflite
**inference-server_1        | /opt/app/acap_runtime/acapruntime: error while loading shared libraries: liblarod.so.1: cannot open shared object file: No such file or directory**
object-detector-python_inference-server_1 exited with code 127
object-detector-python_acap_dl-models_1 exited with code 0
object-detector-python_1  | object-detector-python connect to: inference-server
object-detector-python_1  | <_InactiveRpcError of RPC that terminated with:
object-detector-python_1  | 	status = StatusCode.DEADLINE_EXCEEDED
object-detector-python_1  | 	details = "Deadline Exceeded"
object-detector-python_1  | 	debug_error_string = "{"created":"@1673593775.919770280","description":"Deadline Exceeded","file":"src/core/ext/filters/deadline/deadline_filter.cc","file_line":81,"grpc_status":4}"

Is it because of any access permission to the files ? Thank you in advance

**Reply by Corallo** [reply]
This is our fault. The static-image.yml was not updated.
We'll merge the fix soon. In the meantime, you can take a look at the [PR](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/pull/128) to see what need to be changed, or pull the branch [static-image-fix](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/static-image-fix).

Good that you spotted it!

**Comment by param-esec** [comment]
I did `docker save axisecp/acap-runtime:latest-aarch64-containerized | docker --host tcp://$DEVICE_IP:$DOCKER_PORT load`
but i am still getting the same error when i run this command `docker-compose --host tcp://$DEVICE_IP:$DOCKER_PORT --env-file 
./config/env.$ARCH.$CHIP up`

error:
[+] Running 1/1
 ✘ inference-server Error                                                                                          0.7s
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on [::1]:53: read udp [::1]:54571->[::1]:53: read: connection refused

**Reply by Corallo** [reply]
Can you provide more log? I can't see what docker is trying to get from the docker regestry
Instead of pulling the latest acap-runtime, you might want to check exactly which version it is looking for.

**Reply by param-esec** [reply]
![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/b0b0b28e-692c-4a3c-bcd5-464d6f7b168e)

to get info of the container loaded- `docker image inspect axisecp/acap-runtime:latest-aarch64-containerized`
![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/6c6c904d-e9be-41b2-b1d3-1941bcb8040b)

**Reply by Corallo** [reply]
Okay, then verify with 
`docker --host tcp://$DEVICE_IP:$DOCKER_PORT ps `
if the acap runtime and the other two images are there.

If they are there, run
`docker-compose --host tcp://$DEVICE_IP:$DOCKER_PORT --env-file  ./config/env.$ARCH.$CHIP up`
If it still try to connect to the docker regestry, look at which image it is looking for.

**Reply by param-esec** [reply]
![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/0d2678ce-1630-411e-893c-b0145f561a52)

error message on docker daemon:

![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/38df7458-e901-4bb4-bb5b-d5cd6aebe310)
